python/pyspark/ml/stat.py

```diff
@@ -71,6 +71,67 @@ def test(dataset, featuresCol, labelCol):
         return _java2py(sc, javaTestObj.test(*args))
 
 
+class Correlation(object):
+    """
+    .. note:: Experimental
+
+    Compute the correlation matrix for the input dataset of Vectors using the specified method.
+    Methods currently supported: `pearson` (default), `spearman`.
+
+    .. note:: For Spearman, a rank correlation, we need to create an RDD[Double] for each column
```

Comment:
Sorry, I picked up that the doc gen will fail here - there needs to be 2 spaces before the start of each subsequent line, like this:

```
.. note:: For Spearman, a rank correlation, we need to create an RDD[Double] for each column
  and sort it in order to retrieve the ranks and then join the columns back into an RDD[Vector],
  which is fairly costly. Cache the input Dataset before calling corr with `method = 'spearman'`
  to avoid recomputing the common lineage.
```